conda_build/build.py

```diff
@@ -236,7 +236,10 @@ def detect_and_record_prefix_files(m, files, prefix, config):
         else:
             files_with_prefix = []
 
-    if files_with_prefix and not m.get_value('build/noarch_python'):
+    is_noarch = m.get_value('build/noarch_python') or \
+        str(m.get_value('build/noarch')).lower() == "python"
```

Comment:
Stylistically, I prefer to start the statement after = with a (, so that the \ line continuation is unnecessary.  I find that this prevents errors later when editing these compound statements.